File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> hMap= new HashMap <>();
        
        for(String word:strs){
            char[] chars=word.toCharArray();
            Arrays.sort(chars);
            String sortedWord= new String (chars);
            
            if(!hMap.containsKey(sortedWord)){
                hMap.put(sortedWord,new ArrayList<>());
            }
            hMap.get(sortedWord).add(word);
        }
        return new ArrayList<>(hMap.values());
    }   
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n =strs.size();
        
        unordered_map<string,vector<string>>mp;
        
        for (int i=0;i<n;i++){
            string str=strs[i];
            sort(strs[i].begin(),strs[i].end());
            
            mp[strs[i]].push_back(str);
            
            
        }
        vector<vector<string>>result;
        for (auto v:mp){
            result.push_back(v.second);
        }
        
        return result;
    }
};